CRM/ManualDirectDebit/Hook/Post/Contribution.php

```diff
@@ -1,21 +1,114 @@
 <?php
 
 use CRM_ManualDirectDebit_Common_CollectionReminderSendFlagManager as CollectionReminderSendFlagManager;
+use CRM_ManualDirectDebit_Hook_Custom_Contribution_ContributionDataGenerator as ContributionDataGenerator;
+use CRM_ManualDirectDebit_Common_MandateStorageManager as MandateStorageManager;
 
 class CRM_ManualDirectDebit_Hook_Post_Contribution {
 
   private $contributionId;
+  private $contactId;
+  private $contributionRecurId;
+  private $mandateContributionConnector;
 
-  public function __construct($contributionId) {
+  public function __construct($contributionId, $contactId, $contributionRecurId) {
     $this->contributionId = $contributionId;
+    $this->contactId = $contactId;
+    $this->contributionRecurId = $contributionRecurId;
+    $this->mandateContributionConnector = CRM_ManualDirectDebit_Hook_MandateContributionConnector::getInstance();
   }
 
   public function process() {
     $this->createCollectionReminderSendFlag();
+    $this->processTheWebformSubmission();
+  }
+
+  /**
+   * Processes the webform submission.
+   */
+  private function processTheWebformSubmission() {
```

Comment:
The existing code is not here to only support webforms, but it should also support direct debit submissions from admin form or autorenewal contexts, which does not seem to be the case with this PR updates.

Reply:
I have reverted the DataGenerator class changes.